### DISCUSSION TITLE
Offline support for PWAs

### DISCUSSION THREAD

**Discussion by metmarkosaric** [plausible/analytics#300] [Anything Else]
"Queue up events/page changes when the PWAs are offline and then send them all when network connectivity is restored"

(feedback via email)

**Comment by Lucent** [comment]
Happy for any kind of detection of standalone mode to track PWA usage.

**Reply by Booligoosh** [reply]
To track PWA usage, you can just add a utm_source=pwa query param to your start_url in the manifest. Works across iOS, Android and desktop PWAs.

(also, thanks for making PTable!!)

**Comment by bzoum** [comment]
Hello,
I would be interested as well to track the use of the PWA. In particular, I would be interested by detecting when the visitors are on the app or on their usual browser.

**Reply by Booligoosh** [reply]
Copied from my comment in the previous thread:

To track PWA usage, you can just add a utm_source=pwa query param to your start_url in the manifest. Works across iOS, Android and desktop PWAs.

**Comment by rosano** [comment]
I noticed this link in the docs https://plausible.io/docs/integration-guides#plugin-that-adds-dnt-and-offline-functionality

Is that a current workaround for offline support?

**Reply by metmarkosaric** [reply]
hi @rosano! we haven't tested it but yeah, it should work as advertised.
